### DISCUSSION TITLE
mkdocs not producing python code documentation

### DISCUSSION THREAD

**Discussion by vkuehn** [mkdocs/mkdocs#3608] [Q&A] [answered]
It has been asked a couple of times on stackoverflow and I have the same issue.

with that I try to generate documentation from docstrings

<pre>
plugins:
  - search
  - mkdocstrings:
      handlers:
        python:
          setup_commands:
            - import sys
            - sys.path.append('../')
            - print("system path ----------------------->" + str(sys.path))
          selection:
            new_path_syntax: true`
</pre>

 
with mkdocs build -v I can not see the print but I can see that griffe picked up the modules.
strangely can't forward the output to a text file with >

with griffe dump MODULENAME I can see the docstrings are picked up.

Still mkdocs is not producing any module documentation.

mkdocs version is 1.5.3

**Comment by pawamoy** [comment]
Hello.

Any output coming from the setup commands is discarded, because the legacy Python handler reads the output of pytkdocs and loads it as JSON. Your print statement would break that if we didn't discard it.

Not sure what you mean about forwarding to a file: these are Python statements, not shell commands, so `>` is an operator here, not the shell redirection symbol.

About not seeing any API docs generated, I invite you to open an issue in https://github.com/mkdocstrings/python-legacy. But before that: have you tried the [new handler](https://github.com/mkdocstrings/python)? It's much better than the legacy one :)

**Comment by vkuehn** [comment]
Hello,

usually I can write log output that way to text files.

Not really sure how to diasable the old handler and use the new one.
I have that in poetry.lock
<pre>
[package.extras]
docs = ["mdx-gh-links (>=0.2)", "mkdocs (>=1.5)", "mkdocs-gen-files", "mkdocs-literate-nav", "mkdocs-nature (>=0.6)", "mkdocs-section-index", "mkdocstrings[python]"]
testing = ["coverage", "pyyaml"]
</pre>

and in a make file call it like
<pre>
.PHONY: docs-test
docs-test: ## Test if documentation can be built without warnings or errors
	@poetry run mkdocs build -s
</pre>

anything else to be done ?

**Comment by pawamoy** [comment]
> usually I can write log output that way to text files.

In shell scripts yes, not in Python :slightly_smiling_face: 

---

With `mkdocstrings[python]` in your dependencies, and a recent lock file, this should install the new handler :+1: 
If that's the case, then the options you are using are outdated (they are for the legacy handler).

Try this instead:

```yaml
plugins:
  - search
  - mkdocstrings:
      handlers:
        python:
          paths: [..]
```

See the docs for the `paths` option: https://mkdocstrings.github.io/python/usage/#finding-modules.

I'll move this issue to a discussion since there's nothing actionable for MkDocs.

**Comment by vkuehn** [comment]
if I do that I get this error

<pre>
ERROR   -  Error reading page 'modules.md': 'python_try'
...
KeyError: 'python_try'
</pre>

the project structure looks like

```
python_try
├── codecov.yaml
├── docs
│   ├── index.md
│   └── modules.md
├── LICENSE
├── Makefile
├── mkdocs.yml
├── poetry.lock
├── poetry.toml
├── pyproject.toml
├── python_try
│   ├── foo.py
│   ├── __init__.py
├── README.md
├── requierements.txt
├── site
├── tests
│   └── test_foo.py
└── tox.ini
```

so according to the documentation I have changed the path setting to 

paths: [python_try]

which works but than the modules page is empty

**Reply by vkuehn** [reply]
@pawamoy  if you kindly can comment on the last post ?

**Reply by pawamoy** [reply]
> so according to the documentation I have changed the path setting to `paths: [python_try]`

That's not right.

Your are in the first documented case:

```
📁 root/
├── 📄 mkdocs.yml
└── 📁 python_try/
```

Therefore you must either set `paths: [.]` or remove the `paths` setting.

```
plugins:
- mkdocstrings:
    handlers:
      python:
        paths: [.]  # actually not needed, default
```

**Reply by vkuehn** [reply]
Thanks @pawamoy that's one of the things I tried before asking here and still doesn't

I have published the [reproduceable example](https://github.com/vkuehn/python-try). 

sorry for the pain but would appreciate if mkdocs can be used for code documentation as sphinxs has it's own flaws

**Reply by pawamoy** [reply, marked as answer]
See https://mkdocstrings.github.io/python/usage/configuration/members/#show_submodules to inject docs for submodules recursively :)

**Reply by vkuehn** [reply]
@pawamoy thanks that did work nicely.  A bit confusing for newbees until here. However very happy it works

**Reply by pawamoy** [reply]
Thanks for the feedback, I'll try to make this more prominent in the docs :slightly_smiling_face:
